### PR TITLE
Testfx sketch on map

### PR DESCRIPTION
**Pull request by JonLavi**

### CODE REVIEW

File: display_information/sketch-on-map/build.gradle [hunk 1]
```diff
@@ -34,6 +34,10 @@ dependencies {
     compile "com.esri.arcgisruntime:arcgis-java:$arcgisVersion"
     natives "com.esri.arcgisruntime:arcgis-java-jnilibs:$arcgisVersion"
     natives "com.esri.arcgisruntime:arcgis-java-resources:$arcgisVersion"
+    testCompile "junit:junit:4.13-beta-3"
```
Comment (tschie) [inline display_information/sketch-on-map/build.gradle]: I think all you need is
```groovy
testImplementation "org.testfx:testfx-core:4.0.16-alpha"
testImplementation "org.testfx:testfx-junit:4.0.16-alpha"
```

File: display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+  public void setup() {
+
+    // get a handle on the MapView and GraphicsOverlay
+    Node mapViewNode = lookup("#mapView").query();
+    if (mapViewNode instanceof MapView) {
+      mapView = (MapView) mapViewNode;
+      graphicsOverlay = mapView.getGraphicsOverlays().get(0);
+
+    } else {
+      fail("MapView Node could not be found");
```
Comment (tschie) [inline display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java]: I think TestFX will thrown an exception if the node isn't found. So no need for `fail`.

File: display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    saveButton = lookup("#saveButton").queryButton();
+    clearButton = lookup("#clearButton").queryButton();
+  }
+
+  /**
+   * Test Case 1: Use the SketchEditor to draw a point on the map, and verify that it is added to the GraphicsOverlay.
+   */
+  @Test
+  public void drawPoint() {
+    clickOn(pointButton);
```
Comment (tschie) [inline display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java]: You can chain these
```java
clickOn(pointButton)
   .clickOn(mapView)
   .clickOn(saveButton);
```

That works nicely when you might have async actions that need to happen after the other has finished.

File: display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+    clickOn(mapView);
+    clickOn(saveButton);
+
+    ListenableList<Graphic> graphics = graphicsOverlay.getGraphics();
+    assertEquals(1, graphics.size());
+    assertEquals(GeometryType.POINT, graphics.get(0).getGeometry().getGeometryType());
```
Comment (tschie) [inline display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java]: You might also want to verify the point is in the correct location

File: display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+    double yMin = graphic.getGeometry().getExtent().getYMin();
+    Point2D originaBottomLeftCorner = mapView.locationToScreen(new Point(xMin, yMin, mapView.getSpatialReference()));
+    moveTo(mapView, Pos.TOP_LEFT, originaBottomLeftCorner, Motion.DIRECT);
+    clickOn(MouseButton.PRIMARY);
+
+    // start editing
+    clickOn(editButton);
+
+    // move to the bottom left vertex, select it, and move it
+    moveTo(mapView, Pos.TOP_LEFT, originaBottomLeftCorner, Motion.DIRECT);
```
Comment (tschie) [inline display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java]: Seem to be a few different ways of dragging going on

**Review by tschie — COMMENTED** [review]
Looking good so far!

File: display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java (hunk 4 above)
Comment (JonLavi) [reply display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java]: Yep, added this. Would it be ok to leave this out of the remainder of the tests, seeing that if this passes it proves that SketchEditor translates the screen point to the map point correctly?

File: display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java (hunk 3 above)
Comment (JonLavi) [reply display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java]: Yes, good point. I can also change the methods `clickOnFourPoints()` and `clickAndDragFourPoints()` to:
```
private FxRobotInterface clickOnFourPoints() {
    return clickOn(mapView).[...]
```
that way we can chain other actions after them. However it won't work for chaining the custom methods themselves onto built-in FxRobot methods.. What do you think?

File: display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java (hunk 5 above)
Comment (JonLavi) [reply display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java]: The way it was just there, it's actually adding an additional vertex, not moving an existing one. I hadn't realise it's click and drag to move a vertex, not click and click.
Updated to actually drag the existing vertex somewhere else.

**Comment by JonLavi** [comment]
closing as there will not likely be any progress on this in the forseable future

**Pull request closed without merging**